### PR TITLE
[YUNIKORN-1222] [SI] preempt pods based on the priorities

### PR DESCRIPTION
**Pull request by manirajv06**
### What is this PR for?

SI changes for https://issues.apache.org/jira/browse/YUNIKORN-1184

### What type of PR is it?
* [ ] - Feature

### Todos
* [ ] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-1222

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

**Comment by craigcondit** [comment]
@manirajv06 please regenerate the SI code (run `make` and commit `lib/go/si/si.pb.go`.

File: scheduler-interface-spec.md [hunk 1]
```diff
@@ -447,6 +447,8 @@ message AllocationAsk {
   // Is this a placeholder ask (true) or a real ask (false), defaults to false
   // ignored if the taskGroupName is not set
   bool placeholder = 10;
+  // Does this ask originates the application?
```
Comment (craigcondit) [inline scheduler-interface-spec.md:450]: Grammar: should be `Is this ask the originator of the application?`

File: scheduler-interface-spec.md [hunk 2]
```diff
@@ -714,6 +716,7 @@ const (
 	KeyPodName      = "podName"
 	KeyNamespace    = "namespace"
 	KeyRequiredNode = "requiredNode"
+	KeyAllowPreemption = "allowPreemption"
```
Comment (craigcondit) [inline scheduler-interface-spec.md:719]: Update spacing of this block.

**Review by craigcondit — CHANGES_REQUESTED** [review]

**Review by wilfred-s — APPROVED** [review]
LGTM

**Review by wilfred-s — CHANGES_REQUESTED** [review]
Missing two files in the commit: `si.proto` and `lib/go/common/constants.go`
Re-run `make` to generate the files

**Comment by manirajv06** [comment]
> Missing two files in the commit: `si.proto` and `lib/go/common/constants.go` Re-run `make` to generate the files



> Missing two files in the commit: `si.proto` and `lib/go/common/constants.go` Re-run `make` to generate the files

added those two files

**Review by wilfred-s — APPROVED** [review]
LGTM

**Pull request closed without merging**